controls/postgres_spec.rb

```diff
@@ -157,11 +157,24 @@
   impact 1.0
   title 'Use salted MD5 to store postgresql passwords'
   desc 'Store postgresql passwords in salted hash format (e.g. salted MD5).'
-  describe postgres_session(USER, PASSWORD).query('SELECT passwd FROM pg_shadow;') do
-    its('output') { should match(/^md5\S*$/) }
-  end
-  describe postgres_conf(POSTGRES_CONF_PATH) do
-    its('password_encryption') { should eq 'on' }
+  describe command('psql -V') do
```

Comment:
Can we make this a ruby function? e.g. `def pg_version()` that returns 9 or 10 as the version? Then the test could look much simpler:
```
expected_passwd = /^md5\S*$/
expected_password_encryption = 'on'

if pg_version() == 10
  expected_passwd = /^scram-sha-256\S*$/
  expected_password_encryption = 'scram-sha-256'
end

describe postgres_session(USER, PASSWORD).query('SELECT passwd FROM pg_shadow;') do
    its('output') { should match(expected_passwd) }
end
describe postgres_conf(POSTGRES_CONF_PATH) do
    its('password_encryption') { should eq expected_password_encryption }
end
```

Reply:
@chris-rock I'm afraid my knowledge of Ruby is asymptotic to zero :)